src/Microsoft.AspNet.Authentication.OAuth/OAuthHandler.cs

```diff
@@ -89,7 +88,8 @@ public OAuthHandler(HttpClient backchannel)
                 var value = query["error"];
                 if (!StringValues.IsNullOrEmpty(value))
                 {
-                    Logger.LogVerbose("Remote server returned an error: " + Request.QueryString);
+                    Logger.LogWarning("Remote server returned an error: " + value);
```

Comment:
As observed by @sunsided, trying to write to the response stream is even worse, since you get a "headers already sent" exception when the default 404 handler tries to update the response code: https://github.com/aspnet-contrib/AspNet.Security.OpenIdConnect.Server/issues/138#issuecomment-141719610